File: SALARY.py
def check_same(l):
    t=l[0]
    ret=1
    for i in l:
        if i!=t:
            ret-=1
            break
    return ret
def func(l):
    h=1
    turns=0
    while(check_same(l)==0):
        turns+=1
        h=0
        maxe=l.index(max(l))
        for i in range (len(l)):
            if i!=maxe:
                l[i]+=1
    return turns
t=int(input())
res=[]
for i in range(t):
    n=int(input())
    l=list(map(int,input().split()))
    res+=[func(l)]
for i in res:
    print(i)

File: SALARY.cpp
//This code is made by Chandravo Bhattacharya
#include<bits/stdc++.h>
#include<vector>
using namespace std;
int check_same(int *p,int n){
    int temp=*p;
    int ret=1;
    for (int i=0;i<n;i++){
        if (*(p+i)!=temp){
            ret=0;
            break;
        }
    }
    return ret;
}
int func(int *p,int n){
    int turns=0;
    while(check_same(p,n)==0){
        turns++;
        int maxi=distance(p,max_element(p,p+n));
        for (int i=0;i<n;i++){
            if (i!=maxi){
                (*(p+i))++;
            }
        }
    }
    return turns;
}
int main(){
    int t;cin>>t;
    vector<int> res;
    while(t--){
        int n;cin>>n;
        int arr[n];
        for (int i=0;i<n;i++){
            cin>>arr[i];
        }
        int *p=arr;
        int ans=func(p,n);
        res.push_back(ans);
    }
    for (int i:res){
        cout<<i<<endl;
    }   
}